src/ocean/core/SmartUnion.d

```diff
@@ -226,6 +253,34 @@ private struct SmartUnionIntern ( U )
      **************************************************************************/
 
     Active active;
+
+    /***************************************************************************
+
+        List of active state names
+
+    ***************************************************************************/
+
+    const istring[] active_names = member_string_list();
+
+    /***************************************************************************
+
+        CTFE function to generate the list of active state names for union U.
+
+        Returns:
+            a list containing the names of each of the active states of the
+            smart-union (i.e. the names of the fields of U)
+
+    ***************************************************************************/
+
+    static private istring[] member_string_list ( )
+    {
+        istring[] names = ["none"[]];
```

Comment:
I wonder if the empty string would be a better choice here (if the user is allowed to have the `none` as a member)?